dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/NativeHttpRestClient.java

```diff
@@ -0,0 +1,88 @@
+package com.dotcms.content.elasticsearch.util;
+
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import javax.net.ssl.SSLContext;
+import okhttp3.Authenticator;
+import okhttp3.ConnectionPool;
+import okhttp3.Credentials;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.Route;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+
+/**
+ * Custom ES client that connects to the ES server using Http2
+ * @author nollymar
+ */
+public class NativeHttpRestClient {
+
+    private static OkHttpClient client;
+    private static String url;
+
+    public NativeHttpRestClient(final SSLContext sslContext) {
+
+        final String protocol = sslContext != null ? "https://" : "http://";
+
+        client = new OkHttpClient.Builder()
+                .authenticator(new Authenticator() {
+                    @Override public Request authenticate(Route route, Response response) {
+                        if (response.request().header("Authorization") != null) {
+                            return null; // Give up, we've already attempted to authenticate.
+                        }
+
+                        final String credential = Credentials.basic(
+                                Config.getStringProperty("ES_AUTH_BASIC_USER", null),
+                                Config.getStringProperty("ES_AUTH_BASIC_PASSWORD", null));
+
+                        return response.request().newBuilder()
+                                .header("Authorization", credential)
+                                .build();
+                    }
+                })
+                .sslSocketFactory(sslContext.getSocketFactory())
+                .connectionPool(new ConnectionPool(5, 5, TimeUnit.MINUTES))
+                .build();
+
+        NativeHttpRestClient.url =
+                protocol + Config.getStringProperty("ES_HOSTNAME", "127.0.0.1") + ":" + Config
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not hard code the IP address ](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5147396)